Warn in log if user's WP version is too old for Jetpack

gravityrail opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

When I was testing another PR on VVV, the tests would fail with "class Jetpack not found" errors. It turns out that this was because I had an old (5.0-alpha) source tree in my VM, and there's an early return in jetpack.php if it detects an incompatible WordPress version.

My hope is that this change won't result in crazy verbose logging on sites with old versions of Jetpack, but... those sites are running an incompatible WordPress version anyway, so if it wasn't this weirdness it would be other weirdness. It also might be helpful in case they're missing the notices in their wp-admin.

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Warn in logs if Jetpack plugin can't load due to incompatible WordPress version.

gravityrail committed:
Warn in log if user's WP version is too old for Jetpack

kraftbj commented:
I think this would be pretty noisy running on every load, but this is a situation that generally only folks like us would find ourselves in.

WordPress itself won't let users update to versions of Jetpack that are indicated by us as needing a newer version of WordPress, so this would only be hit by folks updating Jetpack in weird ways. 

Until recently we didn't return out at all until some of us developers fataled out when manually installing Jetpack on WP 4.9.

jeherve reviewed (changes requested):
Quick suggestion to avoid flooding too many error logs, and to ensure the message will be translatable.

jeherve committed:
Wrap error logging into WP_DEBUG check

This should avoid flooding folks' error logs; we'll only log it if Debugging mode is on.

See https://github.com/Automattic/jetpack/pull/12341#discussion_r284271003

jeherve reviewed (approved):
This should be good to merge now that we are a bit more conservative, imo.

@kraftbj I'll let you merge if you are happy with this too.
